nodes/logic/switch_MK2.py

```diff
@@ -0,0 +1,165 @@
+    """
+    bl_idname = 'SvSwitchNodeMK2'
+    bl_label = 'Switch mk2'
+    bl_icon = 'ACTION_TWEAK'
+    
+    input_items = [
+            ("True", "True", "", 0),
+            ("False", "False", "", 1),
+            ("None", "None", "", 2)
+        ]
+    
+    def change_sockets(self, context):
+        pre_num = len(self.inputs)//2
+        diff = self.socket_number - pre_num
+        if diff > 0:
+            for i in range(diff):
+                num = pre_num + i
+                name_A = 'A_{}'.format(num)
+                name_B = 'B_{}'.format(num)
+            
+                self.inputs.new("StringsSocket", name_A).prop_name = name_A
+                self.inputs.new("StringsSocket", name_B).prop_name = name_B
+                self.outputs.new("StringsSocket", "Out_{}".format(num))
+            
+        elif diff < 0:
+            for i in range(abs(diff)):
+                self.inputs.remove(self.inputs[-1])
+                self.inputs.remove(self.inputs[-1])
+                self.outputs.remove(self.outputs[-1])
+
+    switch_state  = EnumProperty(items=input_items[:2], name="state", default="False", update=updateNode)
+    socket_number = IntProperty(name="count", min=1, max=10, default=1, update=change_sockets)
+    flatten_list  = BoolProperty(name='Flatten list', description='for matrixes and objects only',
+                                 default=False, update=updateNode)
+        
+    for i in range(10):
+        name_A = 'A_{}'.format(i)
+        name_B = 'B_{}'.format(i)
+        locals()[name_A] = EnumProperty(items=input_items, name=name_A, default="True", update=updateNode)
+        locals()[name_B] = EnumProperty(items=input_items, name=name_B, default="False", update=updateNode)
+    
+    def sv_init(self, context):        
+        self.inputs.new("StringsSocket", "State").prop_name = 'switch_state'
+        self.inputs.new("StringsSocket", "A_0").prop_name   = 'A_0'
+        self.inputs.new("StringsSocket", "B_0").prop_name   = 'B_0'
+        self.outputs.new("StringsSocket", "Out_0")
+    
+    def draw_buttons_ext(self, context, layout):
+        layout.prop(self, 'socket_number', text="in/out number")
+        layout.prop(self, 'flatten_list')
+        
+    def rclick_menu(self, context, layout):
+        layout.prop(self, 'socket_number', text="in/out number")
+        layout.prop(self, 'flatten_list')
+    
+    def update(self):
+        inputs_A = [i for i in self.inputs if i.name[0] == 'A']
+        for in_soc, out_soc in zip(inputs_A, self.outputs):
+            
+            if not in_soc.links:
+                new_type = in_soc.bl_idname
+            else:
+                in_other = get_other_socket(in_soc)
+                new_type = in_other.bl_idname
+                
+            if new_type == out_soc.bl_idname:
+                continue
+            
+            replace_socket(out_soc, new_type)
+        
+    def process(self):
+        
+        def get_bool(soc):
```

Comment:
place `get_bool`, `check_data_in` as normal functions at the top of the file, and consider turning `check_data_out` into a class method